src/client.rs

```diff
@@ -27,83 +27,144 @@ use crate::Readdir;
 /// # Examples
 ///
 /// ```
-/// use hdrs::Client;
+/// use hdrs::{Client, ClientBuilder};
 ///
-/// let fs = Client::connect("default");
+/// let fs = ClientBuilder::new("default").with_user("default").with_kerberos_ticket_cache_path("/tmp/krb5_111").connect();
 /// ```
 #[derive(Debug)]
 pub struct Client {
     fs: hdfsFS,
 }
 
-/// HDFS's client handle is thread safe.
-unsafe impl Send for Client {}
-unsafe impl Sync for Client {}
+/// The builder of connecting to hdfs clusters.
+///
+/// # Examples
+///
+/// ```
+/// use hdrs::ClientBuilder;
+/// use hdrs::Client;
+///
+/// let fs = ClientBuilder::new("default").with_user("default").with_kerberos_ticket_cache_path("/tmp/krb5_111").connect();
+/// ```
+pub struct ClientBuilder {
+    name_node: String,
+    user: Option<String>,
+    kerberos_ticket_cache_path: Option<String>
+}
 
-impl Client {
-    /// Connect to a name node with port
+impl ClientBuilder {
+
+    /// Create a ClientBuilder with name node
     ///
-    /// Returns an [`io::Result`] if any error happens.
+    /// Returns an [`hdrs::ClientBuilder`]
     ///
     /// # Examples
     ///
     /// ```
+    /// use hdrs::ClientBuilder;
     /// use hdrs::Client;
     ///
-    /// let fs = Client::connect("default");
+    /// let builder = ClientBuilder::new("default");
     /// ```
-    pub fn connect(name_node: &str) -> io::Result<Self> {
-        prepare_env()?;
-
-        set_errno(Errno(0));
-
-        debug!("connect name node {}", name_node);
-
-        let fs = unsafe {
-            let name_node = CString::new(name_node)?;
-            hdfsConnect(name_node.as_ptr(), 0)
-        };
-
-        if fs.is_null() {
-            return Err(io::Error::last_os_error());
+    ///
+    pub fn new(name_node: &str) -> ClientBuilder {
+        ClientBuilder {
+            name_node: name_node.to_string(),
+            user: None,
+            kerberos_ticket_cache_path: None
         }
+    }
 
-        debug!("name node {} connected", name_node);
-        Ok(Client { fs })
+    /// Set the user for existing ClientBuilder
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hdrs::ClientBuilder;
+    /// use hdrs::Client;
+    ///
+    /// let mut builder = ClientBuilder::new("default");
+    /// builder.with_user("default");
+    /// let client = builder.connect();
+    /// ```
+    ///
+    pub fn with_user(&mut self, user: &str) -> &mut ClientBuilder {
```

Comment:
Hi, I expect to have API:

```rust
fn with_user(mut self, user: &str) -> ClientBuilder {}
```

And

```rust
pub fn connect(self) -> io::Result<Client>
```

This will make it possible to call our API in chain.